tests/Tokenizer/TokensTest.php

```diff
@@ -1641,6 +1646,9 @@ private function getBlockEdgeCachingTestTokens(): Tokens
         ]);
     }
 
+    /**
+     * @param Tokens::BLOCK_TYPE_* $type
```

Comment:
pls don't, there is no such type as `Tokens::BLOCK_TYPE_*`

some sca tools will try to convert it to
```diff
-private static function assertFindBlockEnd(int $expectedIndex, string $source, int $type, int $searchIndex): void
+private static function assertFindBlockEnd(int $expectedIndex, string $source, Tokens::BLOCK_TYPE_* $type, int $searchIndex): void
```

which will obviously fail

Reply:
```suggestion
     * @param int $type one of Tokens::BLOCK_TYPE_*
```

Reply:
> there is no such type as `Tokens::BLOCK_TYPE_*`

Surely there's no such native PHP type, but neither there is `ArgumentAnalysis[]`.

As for modern static analysis tools, they understand it — here's an example in [PHPStan](https://phpstan.org/r/d602d897-45c5-4ac8-aefc-2a7fb62571de) and [Psalm](https://psalm.dev/r/e72fd86cc8).

> some sca tools will try to convert it to

What tools? That would probably mean they're broken; they should validate that the PHPDoc type is a correct PHP type before making such changes. If they don't do that, `ArgumentAnalysis[]` would also break them.